### ISSUE TITLE
Accessibility: make sure, that the focus of all interactive elements is consistent

### ISSUE THREAD

**Issue by JuliaKirschenheuter** [nextcloud/nextcloud-vue#2824]
This is an accessibility issue.

Some elements on the page have not the same focus styles. There are some examples: 

![image](https://user-images.githubusercontent.com/6078378/178430441-f6a82214-5818-4366-8969-c62f6f3c0885.png)
![image](https://user-images.githubusercontent.com/6078378/178430501-2de950b2-d86b-4752-b1d6-d6d972c055d6.png)
![image](https://user-images.githubusercontent.com/6078378/178430620-7065d9a6-67b0-4180-add8-8bec45bacb72.png)
![image](https://user-images.githubusercontent.com/6078378/178430736-25b2b6c9-c74d-479b-aa98-dab3bac4d9a5.png)
![image](https://user-images.githubusercontent.com/6078378/178430886-54917dab-b1fa-43c2-adb5-7f37e4c0d501.png)
![image](https://user-images.githubusercontent.com/6078378/178431279-8cbe27cb-d5f8-45a3-a647-9dea7647bca0.png)
![image](https://user-images.githubusercontent.com/6078378/178431427-4410bd6e-722c-43de-9c49-44f4725139c7.png)

Try to make sure, that all interactive elements have the same focus on a NC Vue level.

**Comment by Fenn-CS** [comment]
@nextcloud/designers Please, can you suggest a color for the focus? So that the focus color can be defined as a variable in nextcloud core css?

I have seen more use of
```css
outline: 2px solid var(--color-main-text) !important;
```

Example : https://github.com/nextcloud/nextcloud-vue/pull/2899

`--color-main-text` is darkish and I wonder if it matches the theme design. Even with the growing use, there are still places where different colors and border sizes are used for focus hence the need for us to pick a color and introduce two new scss variables for example `--color-action-focus`, `--border-size-focus`

**Comment by jancborchardt** [comment]
@Fenn-CS so Airbnb for example uses black+white so it works on any background, any theme, even images:
![image](https://github.com/nextcloud/nextcloud-vue/assets/925062/10737238-4887-4ed5-8aa8-def9a2075bcb)
This seems best, if possible?

**Comment by jancborchardt** [comment]
@marcoambrosini at least this is what I also remember you suggested? cc @nextcloud/designers too.

**Comment by nimishavijay** [comment]
> Airbnb for example uses black+white so it works on any background, any theme, even images:

I agree! Suggestion: for the colors we use `--color-main-background` in light mode and dark mode (so `#ffffff` and `#181818`) so that it looks seamless in light and dark modes. I also noticed that the border radius for some of the elements is not consistent, it should be a border around the hover feedback. Example: 
![image](https://github.com/nextcloud/nextcloud-vue/assets/52440189/4e2466d9-dfab-438a-b489-5d8ec5be9314)



Also a point to note: The position of the elements should not shift (especially keep in mind because a double border is used here

**Comment by Fenn-CS** [comment]
I saw this and I think it looks great too. (Especially in the sense that the that shade of blue matches with Nextcloud theme color)

![keyboard-focus-design](https://github.com/nextcloud/nextcloud-vue/assets/14317775/472ad55a-3553-4945-a9ed-91c41cbb048f)

Thoughts for `#ffffff` and `#006aa3` (the difference is replacing black with our `--color-primary-element` ???

**Comment by szaimen** [comment]
I would also say that using --color-primary-element as border-line looks much better. WDYT @jancborchardt @nimishavijay

**Comment by raimund-schluessler** [comment]
> I would also say that using --color-primary-element as border-line looks much better. WDYT @jancborchardt @nimishavijay

The problem with `--color-primary-element` is that it is user defined and you cannot rely on it having enough contrast to `#ffffff`. `--color-primary-default` `#0082c9` could be an option, though

**Comment by szaimen** [comment]
I think we now adjust the color if it should be too bright or dark: https://github.com/nextcloud/server/pull/38159
Or is this not enough?

**Comment by marcoambrosini** [comment]
I think that the airbnb way with black and white always guarantees very high visibility and it's probably the best way to go.
To accomplish this without jumping content they add 2 box shadows. We could do the same with --color-main-text inner and --color-main-background outer

<img width="391" alt="Screenshot 2023-06-12 at 16 27 06" src="https://github.com/nextcloud/nextcloud-vue/assets/26852655/53c91f7e-d9f5-4181-a4c1-b06fe69858fa">

**Comment by susnux** [comment]
Should be fixed with out a11y work. They all have the consistent outline
